bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/discovery/SunSynkAccountDiscoveryService.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.sunsynk.internal.discovery;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.openhab.binding.sunsynk.internal.SunSynkBindingConstants;
+import org.openhab.binding.sunsynk.internal.classes.Inverter;
+import org.openhab.binding.sunsynk.internal.handler.SunSynkAccountHandler;
+import org.openhab.binding.sunsynk.internal.handler.SunSynkHandlerFactory;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SunSynkAccountDiscoveryService} is responsible for starting the discovery procedure
+ * that connects to SunSynk Web and imports all registered inverters.
+ * 
+ * @author Lee Charlton - Initial contribution
+ */
+
+public class SunSynkAccountDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(SunSynkAccountDiscoveryService.class);
+
+    private static final int TIMEOUT = 15;
+
+    private final SunSynkAccountHandler handler;
+    private final ThingUID bridgeUID;
+
+    private ScheduledFuture<?> scanTask;
+
+    public SunSynkAccountDiscoveryService(SunSynkAccountHandler handler) {
+        super(SunSynkHandlerFactory.DISCOVERABLE_THING_TYPE_UIDS, TIMEOUT);
+        this.handler = handler;
+        this.bridgeUID = handler.getThing().getUID();
+    }
+
+    private void findInverters() {
+        List<Inverter> inverters = handler.getInvertersFromSunSynk();
+        for (Inverter inverter : inverters) {
+            addThing(inverter);
+        }
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        findInverters();
+    }
+
+    @Override
+    protected void startScan() {
+        if (this.scanTask != null) {
+            scanTask.cancel(true);
+        }
+        this.scanTask = scheduler.schedule(this::findInverters, 0, TimeUnit.SECONDS);
+    }
+
+    @Override
+    protected void stopScan() {
+        super.stopScan();
+
+        if (this.scanTask != null) {
+            this.scanTask.cancel(true);
```

Comment:
```suggestion
        ScheduledFuture<?> scanTask = this.scanTask;
        if (scanTask != null) {
            scanTask.cancel(true);
```

create a local var to prevent null issues.

Reply:
This was part of the skeleton code.
Amended as requested.